src/Guide/Modules/WaitingRoom.cs

```diff
@@ -27,7 +27,9 @@ public async Task AcceptRules([Remainder]string bio = "The user didn't provide t
                 return;
             }
 
-            if(!IsAsciiPrintable(user.Username[0]))
+
+            var visibleName = user.Nickname ?? user.Username;
+            if(!IsAsciiPrintable(visibleName[0]))
```

Comment:
Should be safe enough if we're assuming discord or discord.net would never accidentally allow empty nicknames or names (what if discord.net defaults nicknames to string.Empty instead of null in the future?), in which case this would throw an `IndexOutOfRangeException`.

```suggestion
            if(!string.IsNullOrEmpty(visibleName) && !IsAsciiPrintable(visibleName[0]))
```